openupgradelib/openupgrade_merge_records.py

```diff
@@ -0,0 +1,355 @@
+                len(records), field_name, field.model,
+            )
+
+
+def _adjust_merged_values_orm(env, model_name, record_ids, target_record_id,
+                              field_spec):
+    """"This method deals with the values on the records to be merged +
+    the target record, performing operations that makes sense on the meaning
+    of the model.
+
+    :param field_spec: Dictionary with field names as keys and forced operation
+      to perform as values. If a field is not present here, default operation
+      will be performed.
+
+      Possible operations by field types:
+
+      * Char, Text and Html fields:
+        - 'merge' (default): content is concatenated with an ' | ' as separator
+        - other value: content on target record is preserved
+      * Integer, Float and Monetary fields:
+        - 'sum' (default): Sum all the values of the records.
+        - 'avg': Perform the arithmetic average of the values of the records.
+        - 'max': Put the maximum of all the values.
+        - 'min': Put the minimum of all the values.
+        - other value: content on target record is preserved
+      * Binary field:
+        - any value: content on target record is preserved
+      * Boolean field:
+        - 'and': Perform a logical AND over all values.
+        - 'or': Perform a logical OR over all values.
+        - other value (default): content on target record is preserved
+      * Date and Datetime fields:
+        - 'max': Put the maximum of all the values.
+        - 'min': Put the minimum of all the values.
+        - other value (default): content on target record is preserved
+      * Many2one fields:
+        - any value: content on target record is preserved
+      * Many2many fields:
+        - 'merge' (default): combine all the values
+        - other value: content on target record is preserved
+      * One2many fields:
+        - 'merge' (default): combine all the values
+        - other value: content on target record is preserved
+      * Reference fields:
+        - any value: content on target record is preserved
+      * Selection fields:
+        - any value: content on target record is preserved
+    """
+    model = env[model_name]
+    fields = model._fields.values()
+    all_records = model.browse(record_ids + [target_record_id])
+    target_record = model.browse(target_record_id)
+    vals = {}
+    o2m_changes = 0
+    for field in fields:
+        op = field_spec.get(field.name, False)
+        l = all_records.mapped(field.name)
+        if field.type in ('char', 'text', 'html'):
+            op = op or 'merge'
+            if op == 'merge':
+                vals[field.name] = ' | '.join(l)
+        elif field.type in ('integer', 'float', 'monetary'):
+            op = op or 'sum'
+            if op == 'sum':
+                vals[field.name] = sum(l)
+            elif op == 'avg':
+                vals[field.name] = sum(l) / len(l)
```

Comment:
Should the result be rounded in case of type='integer'?

Reply:
It's handled automatically by ORM.